v0/destinations/posthog/transform.js

```diff
@@ -0,0 +1,137 @@
+const { EventType } = require("../../../constants");
+const {
+  DEFAULT_BASE_ENDPOINT,
+  CONFIG_CATEGORIES,
+  MAPPING_CONFIG
+} = require("./config");
+const {
+  defaultRequestConfig,
+  getBrowserInfo,
+  getDeviceModel,
+  constructPayload,
+  defaultPostRequestConfig,
+  ErrorMessage
+} = require("../../util");
+
+const isValidUrl = url => {
+  try {
+    // eslint-disable-next-line no-new
+    new URL(url);
+    return true;
+  } catch (err) {
+    return false;
+  }
+};
+
+// Logic To match destination Property key that is in Rudder Stack Properties Object.
+const generatePropertyDefination = message => {
+  const PHPropertyJson = CONFIG_CATEGORIES.PROPERTY.name;
+  let propertyJson = MAPPING_CONFIG[PHPropertyJson];
+  let data = {};
+
+  if (message.channel === "mobile") {
+    propertyJson = propertyJson.filter(d => {
+      return d.isScreen || d.all;
+    });
+  } else {
+    propertyJson = propertyJson.filter(d => {
+      return !d.isScreen || d.all;
+    });
+  }
+
+  if (
+    message.channel === "web" &&
+    message.context &&
+    message.context.userAgent
+  ) {
+    const browser = getBrowserInfo(message.context.userAgent);
+    const osInfo = getDeviceModel(message);
+    data.$os = osInfo;
+    data.$browser = browser.name;
+    data.$browser_version = browser.version;
+  }
```

Comment:
Do you want to do it after populating the `data` variable using `constructPayload`?

Reply:
This seems like outdated code. Modified this line of code to first get data from  `constructPayload` and then populating  data with browser property. This is how I am planning to do::
  data = constructPayload(message, propertyJson);

 ```
 // This logic ensures to get browser info only for payload generated from web.
  if (
    message.channel === "web" &&
    message.context &&
    message.context.userAgent
  ) {
    const browser = getBrowserInfo(message.context.userAgent);
    const osInfo = getDeviceModel(message);
    data.$os = osInfo;
    data.$browser = browser.name;
    data.$browser_version = browser.version;
  }
```